native/membrane_videocompositor_wgpu/src/compositor.rs

```diff
@@ -1,14 +1,64 @@
+use std::collections::BTreeMap;
+
 mod textures;
 mod videos;
 
 use textures::*;
 use videos::*;
 
+use crate::errors::CompositorError;
+
+/// A point in 2D space
+pub struct Point(pub f32, pub f32);
```

Comment:
Maybe we could add indexing `Point` like: `point.x` `point.y`? I think it looks better and is not that hard to implement. I've done something like this on `video_transforations` branch in OpenGL and calls like `self.placement.top_left.x` in my opinion look better than `self.placement.top_left.0`.
Something like this should work:
```
/// A point in 2D space
pub struct Point {
    pub x: f32,
    pub y: f32
}
```

Reply:
BTW can we make it more generic and allow other numerical data types in the `Point` struct such as `u32`. I used `Point` while implementing video transformations in OpenGL and using other types seemed to be a more straightforward solution rather than converting back and forth between `f32` and desired type. [Here is a great example.](https://doc.rust-lang.org/book/ch10-01-syntax.html)